TUMFARSynchrony/experimental-hub
# Backend Filters Data API

This PR adds two new API calls. With those API calls, data from filters can be fetched.

## GET_FILTERS_DATA

### Description

Get all data for requested filter for all participants or only specific (if request is made from Participant). Server will respond with all data of filter that match filter name, channel and id.
BE CAUTIOUS: call from participant does not work (look at issues)

### Request

| Type   | Data                             |
| ------ | -------------------------------- |
| `GET_FILTERS_DATA` | [GetFiltersDataRequest](https://github.com/TUMFARSynchrony/experimental-hub/wiki/Data-Types#getfiltersdatarequest) JSON |

### Response

| Type    | Data                           |
| ------- | ------------------------------ |
| `FILTERS_DATA` | dict of \[str, [FiltersData](https://github.com/TUMFARSynchrony/experimental-hub/wiki/Data-Types#filtersdata)\] JSON |

<br>

## GET_FILTERS_TEST_STATUS

### Description

Get all test status for requested filter for all participants or only specific. Server will respond with status of all filters that match filter name, channel and id.

### Request

| Type   | Data                             |
| ------ | -------------------------------- |
| `GET_FILTERS_TEST_STATUS` | [GetFiltersTestStatusRequest](https://github.com/TUMFARSynchrony/experimental-hub/wiki/Data-Types#getfiltersteststatusrequest) JSON |

### Response

| Type    | Data                           |
| ------- | ------------------------------ |
| `FILTERS_TEST_STATUS` | dict of \[str, [FiltersData](https://github.com/TUMFARSynchrony/experimental-hub/wiki/Data-Types#filtersdata)\] JSON |



-----------------------------------------------------------------------------------------
## How to Test:
- go to ConnectionTest page
- join as experimenter and as participant
- on experimenter page:
  - press button `GET_FILTERS_DATA` -> should return empty array
  - press button `GET_FILTERS_TEST_STATUS` -> sends response to participant, go to participant page, there should be an answer with empty array
- add `simple_glasses_detection` filter
- on experimenter page:
  - press button `GET_FILTERS_DATA` -> should return array with glasses_detection to true/false
  - press button `GET_FILTERS_TEST_STATUS` -> sends response to participant, go to participant page, there should be an answer similar to `GET_FILTERS_DATA` 

### Optional
- in ConnectionTest.tsx: change api call to backend
  - remove parameters (filter_id, filter_name, etc.)
  - change name of parameters (filter_name -> filter_nane)
- after each change: look if API call still works (need to refresh page or start new connection)
    
### Example of correct response:
```json
{
    "a": {
        "video": [
            {
                "id": "simple-glasses-detection",
                "data": {
                    "Glasses Detected": false
                }
            }
        ],
        "audio": []
    }
}
```
